packages/engine-test-utils/src/__tests__/engine-server/EngineEvents.spec.ts

```diff
@@ -0,0 +1,404 @@
+                    expect(updateEntry.note.children.length).toEqual(3);
+                    expect(
+                      updateEntry.note.children.find((name) => name === "alpha")
+                    ).toBeTruthy();
+                  } else if (entry.status === "create") {
+                    expect(entry.note.fname).toEqual("alpha");
+                  }
+                });
+
+                done();
+              } catch (err) {
+                done(err);
+              }
+            }
+          );
+
+          engineClient.writeNote(newNote);
+        },
+        {
+          expect,
+          preSetupHook: ENGINE_HOOKS.setupBasic,
+          createEngine: createEngineFromServer,
+        }
+      );
+    });
+  });
+
+  describe("WHEN writing a new note as a grandchild, whose parent doesn't exist yet", () => {
+    test("THEN create event fired for both child and parent (with stub), and update note fired for grandparent", async (done) => {
+      await runEngineTestV5(
+        async ({ engine, vaults, wsRoot }) => {
+          const engineClient = engine as DendronEngineClient;
+
+          const newNote = await NoteTestUtilsV4.createNote({
+            fname: "bar.child.grandchild",
+            vault: vaults[0],
+            wsRoot,
+          });
+
+          engineClient.onEngineNoteStateChanged(
+            (noteChangeEntries: NoteChangeEntry[]) => {
+              try {
+                expect(noteChangeEntries.length).toEqual(3);
+
+                noteChangeEntries.forEach((entry) => {
+                  if (entry.status === "update") {
+                    const updateEntry = entry as NoteChangeUpdateEntry;
+                    expect(updateEntry.note.fname).toEqual("bar");
+                    expect(updateEntry.prevNote.children.length).toEqual(0);
+                    expect(updateEntry.note.children.length).toEqual(1);
+                  } else if (entry.status === "create") {
+                    if (entry.note.fname === "bar.child") {
+                      expect(entry.note.stub).toBeTruthy();
+                    } else if (entry.note.fname === "bar.child.grandchild") {
+                      expect(entry.note.stub).toBeFalsy();
+                    } else {
+                      done({
+                        message: `Unexpected note created with id: ${entry.note.fname}`,
+                      });
+                    }
+                  }
+                });
+
+                done();
+              } catch (err) {
+                done(err);
+              }
+            }
+          );
+
+          engineClient.writeNote(newNote);
+        },
+        {
+          expect,
+          preSetupHook: ENGINE_HOOKS.setupBasic,
+          createEngine: createEngineFromServer,
+        }
+      );
+    });
+  });
+
+  // TODO: This scenario doesn't seem to work - parent is updated, but children appear to get orphaned.
+  describe.skip("WHEN writing over an existing note by changing its ID", () => {
```

Comment:
you need to pass in `updateExisting: true` as the second option

Reply:
I see - should `updateExisting: true` be the default behavior?  Which use cases have `updateExisting: false`?

Reply:
Created [[writeNote Orphans Child Nodes|dendron://private/task.2022.02.08.writenote-orphans-child-nodes]]